diff --git a/include/linux/compiler-gcc.h b/include/linux/compiler-gcc.h
index e2c7f43..901c1cc 100644
--- a/include/linux/compiler-gcc.h
+++ b/include/linux/compiler-gcc.h
@@ -93,10 +93,6 @@
 #define __weak		__attribute__((weak))
 #define __alias(symbol)	__attribute__((alias(#symbol)))
 
-#ifdef RETPOLINE
-#define __noretpoline __attribute__((indirect_branch("keep")))
-#endif
-
 /*
  * it doesn't make sense on ARM (currently the only user of __naked)
  * to trace naked functions because then mcount is called without
