diff --git a/drivers/media/test-drivers/vidtv/vidtv_bridge.c b/drivers/media/test-drivers/vidtv/vidtv_bridge.c
index 74b054947bbe..8b67d883c651 100644
--- a/drivers/media/test-drivers/vidtv/vidtv_bridge.c
+++ b/drivers/media/test-drivers/vidtv/vidtv_bridge.c
@@ -196,7 +196,7 @@ static int vidtv_stop_streaming(struct vidtv_dvb *dvb)
 	vidtv_mux_destroy(dvb->mux);
 	dvb->mux = NULL;
 
-	dev_dbg_ratelimited(dev, "Stopped streaming\n");
+	dev_dbg(dev, "Stopped streaming\n");
 	return 0;
 }
 
