--- linux-3.17.8/include/linux/dma-buf.h	2015-01-08 19:28:01.000000000 +0100
+++ linux-3.16.81/include/linux/dma-buf.h	2020-01-11 03:05:08.000000000 +0100
@@ -30,8 +30,6 @@
 #include <linux/list.h>
 #include <linux/dma-mapping.h>
 #include <linux/fs.h>
-#include <linux/fence.h>
-#include <linux/wait.h>
 
 struct device;
 struct dma_buf;
@@ -117,7 +115,6 @@
  * @exp_name: name of the exporter; useful for debugging.
  * @list_node: node for dma_buf accounting and debugging.
  * @priv: exporter specific private data for this buffer object.
- * @resv: reservation object linked to this dma-buf
  */
 struct dma_buf {
 	size_t size;
@@ -131,17 +128,6 @@
 	const char *exp_name;
 	struct list_head list_node;
 	void *priv;
-	struct reservation_object *resv;
-
-	/* poll support */
-	wait_queue_head_t poll;
-
-	struct dma_buf_poll_cb_t {
-		struct fence_cb cb;
-		wait_queue_head_t *poll;
-
-		unsigned long active;
-	} cb_excl, cb_shared;
 };
 
 /**
@@ -182,11 +168,10 @@
 				struct dma_buf_attachment *dmabuf_attach);
 
 struct dma_buf *dma_buf_export_named(void *priv, const struct dma_buf_ops *ops,
-			       size_t size, int flags, const char *,
-			       struct reservation_object *);
+			       size_t size, int flags, const char *);
 
-#define dma_buf_export(priv, ops, size, flags, resv)	\
-	dma_buf_export_named(priv, ops, size, flags, KBUILD_MODNAME, resv)
+#define dma_buf_export(priv, ops, size, flags)	\
+	dma_buf_export_named(priv, ops, size, flags, KBUILD_MODNAME)
 
 int dma_buf_fd(struct dma_buf *dmabuf, int flags);
 struct dma_buf *dma_buf_get(int fd);
