diff -u linux-4.19.149/drivers/media/common/videobuf2/videobuf2-dma-contig.c linux-4.18.20/drivers/media/common/videobuf2/videobuf2-dma-contig.c
--- linux-4.19.149/drivers/media/common/videobuf2/videobuf2-dma-contig.c	2020-10-01 13:14:54.000000000 +0200
+++ linux-4.18.20/drivers/media/common/videobuf2/videobuf2-dma-contig.c	2018-11-21 09:22:14.000000000 +0100
@@ -222,7 +222,7 @@
 	enum dma_data_direction dma_dir;
 };
 
-static int vb2_dc_dmabuf_ops_attach(struct dma_buf *dbuf,
+static int vb2_dc_dmabuf_ops_attach(struct dma_buf *dbuf, struct device *dev,
 	struct dma_buf_attachment *dbuf_attach)
 {
 	struct vb2_dc_attachment *attach;
@@ -358,6 +358,7 @@
 	.map_dma_buf = vb2_dc_dmabuf_ops_map,
 	.unmap_dma_buf = vb2_dc_dmabuf_ops_unmap,
 	.map = vb2_dc_dmabuf_ops_kmap,
+	.map_atomic = vb2_dc_dmabuf_ops_kmap,
 	.vmap = vb2_dc_dmabuf_ops_vmap,
 	.mmap = vb2_dc_dmabuf_ops_mmap,
 	.release = vb2_dc_dmabuf_ops_release,
diff -u linux-4.19.149/drivers/media/common/videobuf2/videobuf2-dma-sg.c linux-4.18.20/drivers/media/common/videobuf2/videobuf2-dma-sg.c
--- linux-4.19.149/drivers/media/common/videobuf2/videobuf2-dma-sg.c	2020-10-01 13:14:54.000000000 +0200
+++ linux-4.18.20/drivers/media/common/videobuf2/videobuf2-dma-sg.c	2018-11-21 09:22:14.000000000 +0100
@@ -371,7 +371,7 @@
 	enum dma_data_direction dma_dir;
 };
 
-static int vb2_dma_sg_dmabuf_ops_attach(struct dma_buf *dbuf,
+static int vb2_dma_sg_dmabuf_ops_attach(struct dma_buf *dbuf, struct device *dev,
 	struct dma_buf_attachment *dbuf_attach)
 {
 	struct vb2_dma_sg_attachment *attach;
@@ -507,6 +507,7 @@
 	.map_dma_buf = vb2_dma_sg_dmabuf_ops_map,
 	.unmap_dma_buf = vb2_dma_sg_dmabuf_ops_unmap,
 	.map = vb2_dma_sg_dmabuf_ops_kmap,
+	.map_atomic = vb2_dma_sg_dmabuf_ops_kmap,
 	.vmap = vb2_dma_sg_dmabuf_ops_vmap,
 	.mmap = vb2_dma_sg_dmabuf_ops_mmap,
 	.release = vb2_dma_sg_dmabuf_ops_release,
diff -u linux-4.19.149/drivers/media/common/videobuf2/videobuf2-vmalloc.c linux-4.18.20/drivers/media/common/videobuf2/videobuf2-vmalloc.c
--- linux-4.19.149/drivers/media/common/videobuf2/videobuf2-vmalloc.c	2020-10-01 13:14:54.000000000 +0200
+++ linux-4.18.20/drivers/media/common/videobuf2/videobuf2-vmalloc.c	2018-11-21 09:22:14.000000000 +0100
@@ -209,7 +209,7 @@
 	enum dma_data_direction dma_dir;
 };
 
-static int vb2_vmalloc_dmabuf_ops_attach(struct dma_buf *dbuf,
+static int vb2_vmalloc_dmabuf_ops_attach(struct dma_buf *dbuf, struct device *dev,
 	struct dma_buf_attachment *dbuf_attach)
 {
 	struct vb2_vmalloc_attachment *attach;
@@ -346,6 +346,7 @@
 	.map_dma_buf = vb2_vmalloc_dmabuf_ops_map,
 	.unmap_dma_buf = vb2_vmalloc_dmabuf_ops_unmap,
 	.map = vb2_vmalloc_dmabuf_ops_kmap,
+	.map_atomic = vb2_vmalloc_dmabuf_ops_kmap,
 	.vmap = vb2_vmalloc_dmabuf_ops_vmap,
 	.mmap = vb2_vmalloc_dmabuf_ops_mmap,
 	.release = vb2_vmalloc_dmabuf_ops_release,
