--- linux-5.10.1/include/linux/dma-buf.h	2020-12-14 19:33:01.000000000 +0100
+++ linux-5.9.1/include/linux/dma-buf.h	2020-10-17 08:31:22.000000000 +0200
@@ -283,7 +283,6 @@
  * @exp_name: name of the exporter; useful for debugging.
  * @name: userspace-provided name; useful for accounting and debugging,
  *        protected by @resv.
- * @name_lock: spinlock to protect name access
  * @owner: pointer to exporter module; used for refcounting when exporter is a
  *         kernel module.
  * @list_node: node for dma_buf accounting and debugging.
@@ -312,7 +311,7 @@
 	void *vmap_ptr;
 	const char *exp_name;
 	const char *name;
-	spinlock_t name_lock;
+	spinlock_t name_lock; /* spinlock to protect name access */
 	struct module *owner;
 	struct list_head list_node;
 	void *priv;
